kimchi/js/bindings.js

```diff
@@ -2444,8 +2510,35 @@ var caml_pasta_fq_opening_proof_of_rust = function (x) {
   return [0, lr, delta, z1, z2, sg];
 };
 
-// Provides: caml_pasta_fq_commitments_to_rust
+// Provides: caml_fq_lookup_commitments_to_rust
 // Requires: plonk_wasm, caml_pallas_poly_comm_to_rust, js_class_vector_to_rust_vector
+var caml_fq_lookup_commitments_to_rust = function (caml_lc) {
+  var convertArray = function (v) {
+    var n = v.length - 1;
+    var res = new Array(n);
+    for (var i = 0; i < n; ++i) {
+      res[i] = caml_pallas_poly_comm_to_rust(v[i + 1]);
+    }
+    return js_class_vector_to_rust_vector(res);
+  };
+
+  var wasm_sorted = convertArray(caml_lc[1]);
+  var wasm_aggreg = caml_pallas_poly_comm_to_rust(caml_lc[2]);
+  var wasm_runtime;
+  if (caml_lc[3] === 0) {
```

Comment:
Options in jsoo have the following memory representation:
- `None` -> `[0]`
- `Some(x)` -> `[0, x]`.

Reply:
Would it make sense to have some functions/values around that?

Reply:
See https://github.com/o1-labs/snarkyjs-bindings/pull/110#discussion_r1296109899